noteable/datasource_postprocessing.py

```diff
@@ -338,10 +339,12 @@ def postprocess_clickhouse(
 ) -> None:
     connect_args = create_engine_kwargs["connect_args"]
 
-    # Pop the keys we want to use to build the query parameters in the connection string
-    # so that they don't get doubly passed down into the `noteable.sql.connection.Connection.set` call
-    # in `noteable.datasources.bootstrap_datasource()`.
-    dsn_dict["query"] = {
-        "protocol": connect_args.pop("protocol"),
-        "verify": str(connect_args.pop("verify")).lower(), # converts True/False to true/false
-    }
+    # These are the enum options from the JSON schema for the dropdown titled "Secure Connection (HTTPS)"
+    # Convert them to the values that the clickhouse driver expects.
+    verify = connect_args["verify"]
```

Comment:
Probably use a brand new field name in connect args to hold the enum value, then be sure to obliterate it from the dict by time done. Using the same field to represent two different concepts (the enum, vs. sometimes a bool) is gross.